### DISCUSSION TITLE
官网文档-模板渲染ejs无法配置模板，一直报错

### DISCUSSION THREAD

**Discussion by 34892002** [midwayjs/midway#3998] [General / 闲聊]
按照文档走一直报错
注释81行才正常 node_modules\@midwayjs\view\dist\viewManager.js
// assert(filename, `Can't find ${name} from ${config.root.join(',')}`);
而且dev模式修改ejs不会更新，得重新run

**Comment by czy88840616** [comment]
是用了 egg ？

**Comment by 34892002** [comment]
呃，应该没有吧，我按官网文档操作的npm init midway@latest -y，选的 koa-v3 - A web application boilerplate with midway v3(koa)
附上package.json
{
  "name": "my-midway-project",
  "version": "1.0.0",
  "description": "",
  "private": true,
  "dependencies": {
    "@midwayjs/bootstrap": "^3.12.0",
    "@midwayjs/core": "^3.12.0",
    "@midwayjs/info": "^3.12.0",
    "@midwayjs/koa": "^3.12.0",
    "@midwayjs/logger": "^3.1.0",
    "@midwayjs/validate": "^3.12.0"
  },
  "devDependencies": {
    "@midwayjs/mock": "^3.12.0",
    "@midwayjs/view-ejs": "^3.16.2",
    "@types/jest": "^29.2.0",
    "@types/node": "14",
    "cross-env": "^6.0.0",
    "jest": "^29.2.2",
    "mwts": "^1.3.0",
    "mwtsc": "^1.4.0",
    "ts-jest": "^29.0.3",
    "typescript": "~4.8.0"
  },
  "engines": {
    "node": ">=12.0.0"
  },
  "scripts": {
    "start": "NODE_ENV=production node ./bootstrap.js",
    "dev": "cross-env NODE_ENV=local mwtsc --watch --run @midwayjs/mock/app.js",
    "test": "cross-env NODE_ENV=unittest jest",
    "cov": "jest --coverage",
    "lint": "mwts check",
    "lint:fix": "mwts fix",
    "ci": "npm run cov",
    "build": "mwtsc --cleanOutDir"
  },
  "repository": {
    "type": "git",
    "url": ""
  },
  "author": "anonymous",
  "license": "MIT"
}

**Comment by czy88840616** [comment]
![image](https://github.com/user-attachments/assets/50021a56-52da-4ebc-82a8-ad51f8a89f53)
没找到文件?

render 了一个目录不对的文件（比如没写后缀）

**Reply by 34892002** [reply]
我也不知道咋回事，文件路径都是正确的，就是一直报错can't find hello.ejs，然后我就注释了assert这一行就正常渲染了，只是不能热更新了，稍等下我上传一下仓库吧

**Reply by 34892002** [reply]
https://github.com/34892002/ejs-test.git

**Reply by czy88840616** [reply]
![image](https://github.com/user-attachments/assets/4b7676af-d2a7-4d1b-bc61-d9b9aae8ea5a)
看起来没有报错

**Reply by 34892002** [reply]
src\config\config.default.ts 里面的rootDir要注释一下，那个是我加的。配置文档默认是不加这个的

**Reply by czy88840616** [reply]
不加，你的 view 应该放在根目录，而不是 src 里。

**Reply by czy88840616** [reply]
![image](https://github.com/user-attachments/assets/9b765f48-0dcf-4ed4-906c-de87741bf1e8)

**Reply by 34892002** [reply]
擦~眼花了，我该死。抱歉浪费大佬时间了

**Reply by czy88840616** [reply]
缓存试了下也没有问题，用的 dev 带的 local 环境变量，是会让缓存更新的。

**Reply by 34892002** [reply]
抱歉抱歉，是我误打误撞的配置了正确的rootDir，让我产生了注释assert就渲染正常的错觉，功能我又试了下都是正常的
